lib/codecs/src/encoding/format/cef.rs

```diff
@@ -0,0 +1,520 @@
+                actual_length: device_vendor.len(),
+            }
+            .fail()
+            .map_err(|e| e.to_string().into());
+        };
+
+        let device_product = escape_header(self.cef.device_product.as_str());
+        if device_product.len() > DEVICE_PRODUCT_MAX_LENGTH {
+            return ExceededLengthSnafu {
+                field: device_product.clone(),
+                max_length: DEVICE_PRODUCT_MAX_LENGTH,
+                actual_length: device_product.len(),
+            }
+            .fail()
+            .map_err(|e| e.to_string().into());
+        };
+
+        let device_version = escape_header(self.cef.device_version.as_str());
+        if device_version.len() > DEVICE_VERSION_MAX_LENGTH {
+            return ExceededLengthSnafu {
+                field: device_version.clone(),
+                max_length: DEVICE_VERSION_MAX_LENGTH,
+                actual_length: device_version.len(),
+            }
+            .fail()
+            .map_err(|e| e.to_string().into());
+        };
+
+        let device_event_class_id = escape_header(self.cef.device_event_class_id.as_str());
+        if device_event_class_id.len() > DEVICE_EVENT_CLASS_ID_MAX_LENGTH {
+            return ExceededLengthSnafu {
+                field: device_event_class_id.clone(),
+                max_length: DEVICE_EVENT_CLASS_ID_MAX_LENGTH,
+                actual_length: device_event_class_id.len(),
+            }
+            .fail()
+            .map_err(|e| e.to_string().into());
+        };
+
+        for key in self.cef.extensions.keys() {
+            if !key.chars().all(|c| c.is_ascii_alphabetic()) {
+                // TODO: Output all invalid keys
+                return ExtensionNonASCIIKeySnafu { key: key.clone() }
+                    .fail()
+                    .map_err(|e| e.to_string().into());
+            }
+        }
+
+        let device = DeviceSettings::new(
+            device_vendor,
+            device_product,
+            device_version,
+            device_event_class_id,
+        );
+
+        Ok(CefSerializer::new(
+            self.cef.version.clone(),
+            device,
+            self.cef.severity.clone(),
+            self.cef.name.clone(),
+            self.cef.extensions.clone(),
+        ))
+    }
+
+    /// The data type of events that are accepted by `CefSerializer`.
+    pub fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema required by the serializer.
+    pub fn schema_requirement(&self) -> schema::Requirement {
+        // While technically we support `Value` variants that can't be losslessly serialized to
+        // CEF, we don't want to enforce that limitation to users yet.
+        schema::Requirement::empty()
+    }
+}
+
+/// CEF version.
+#[crate::configurable_component]
+#[derive(Debug, Default, Clone)]
+pub enum Version {
+    #[default]
+    /// CEF specification version 0.1.
+    V0,
+    /// CEF specification version 1.x.
+    V1,
+}
+
+impl Version {
```

Comment:
Nit:

If you define this: 
```
impl std::fmt::Display for Version {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", self.as_str())
    }
}
```

Then you can use the instance directly without calling `as_str()` e.g. `format!("{}", version)`.

Reply:
Added